docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,174 @@
+
+We support many functions for Complex Tensors eg. SVD, QR etc. Operations on complex tensors are likely to be
+faster than operations on float tensors mimicking them. We also have a `Vec256 class` for `ComplexFloat` and
+`ComplexDouble` to benefit from Intel assembly instructions for performing vectorized operations on CPU. In
+addition, operations on Complex Tensors utilize specialized kernels for complex provided by LAPACK (for CPU)
+and BLAS (for CUDA) or kernels in `torch` specially written for complex tensors. Thus using functions Complex
+Tensors will provide performance benefits as opposed to users defining their own functions.
+
+.. warning ::
+     Complex Tensors is a beta feature and subject to change.
+
+Creating Complex Tensors
+----------------------
+
+We support two complex dtypes: `torch.cfloat` and `torch.cdouble`
+
+::
+
+     >>> x = torch.randn(2,2, dtype=torch.cfloat)
+     >>> x
+     tensor([[-0.4621-0.0303j, -0.2438-0.5874j],
+          [ 0.7706+0.1421j,  1.2110+0.1918j]])
+
+The default dtype for complex tensors is updated based on the current default tensor type.
+cdouble if double else float.
+
+<add code>
+
+The following factory functions can be used to create complex tensors:
+
+- :func:`torch.tensor`
+- :func:`torch.as_tensor`
+- :func:`torch.empty`
+- :func:`torch.rand`
+- :func:`torch.randn`
+- :func:`torch.as_strided`
+- :func:`torch.from_numpy`
+- :func:`torch.zeros`
+- :func:`torch.ones`
+- :func:`torch.full`
+- :func:`torch.eye`
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real`:
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+
+<1 line description>
+
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+
+Angle and abs
+-------------
+
+<1 line description>
+
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Serialization
+-------------
+
+<1 line description>
+
+::
+
+     >>> torch.save(y, 'complex_tensor.pt')
+     >>> torch.load('complex_tensor.pt')
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+
+Autograd
+--------
+
+PyTorch supports Autograd for Complex Numbers.
```

Comment:
I think the core of this should go in the autograd section and you should link it from here.

Reply:
so should we move everything mentioned here (including the example) to autograd mechanics as well? 
cc. @albanD

Reply:
I think it is already in the autograd doc as well.
The list explaining what you get for different type of functions is good to have here I think.
Not sure what the code sample brings though.